dali/imgcodec/decoders/decoder_test_helper.h

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_IMGCODEC_DECODERS_DECODER_TEST_HELPER_H_
+#define DALI_IMGCODEC_DECODERS_DECODER_TEST_HELPER_H_
+
+#include <string>
+#include <memory>
+#include <vector>
+#include "dali/pipeline/data/tensor.h"
+#include "dali/core/static_switch.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/imgcodec/image_format.h"
+#include "dali/imgcodec/image_decoder.h"
+#include "dali/test/dali_test.h"
+#include "dali/kernels/slice/slice_cpu.h"
+#include "dali/core/stream.h"
+#include "dali/util/file.h"
+#include "dali/util/numpy.h"
+
+namespace dali {
+namespace imgcodec {
+namespace test {
+
+/**
+* @brief Base class template for tests comparing decoder's results with reference images.
+*
+* @tparam OutputType Type, to which the image should be decoded.
+*/
+template<typename OutputType>
+class CpuDecoderTestBase : public ::testing::Test {
+ public:
+  CpuDecoderTestBase() : tp_(4, CPU_ONLY_DEVICE_ID, false, "Decoder test") {}
+
+  /**
+  * @brief Decodes an image and returns the result as a tensor.
+  */
+  Tensor<CPUBackend> Decode(ImageSource *src, const DecodeParams &opts = {}, const ROI &roi = {}) {
+    EXPECT_TRUE(Parser()->CanParse(src));
+    ImageInfo info = Parser()->GetInfo(src);
+
+    Tensor<CPUBackend> result;
+    EXPECT_TRUE(Decoder()->CanDecode(src, opts));
+    TensorShape<> shape = (roi.use_roi() ? roi.shape() : info.shape);
+    result.Resize(shape, type2id<OutputType>::value);
+
+    SampleView<CPUBackend> view(result.raw_mutable_data(), result.shape(), result.type());
+    DecodeResult decode_result = Decoder()->Decode(view, src, opts, roi);
+    EXPECT_TRUE(decode_result.success);
+
+    return result;
+  }
+
+  /**
+  * @brief Checks if two tensors are equal.
+  */
+  void AssertEqual(const Tensor<CPUBackend> &img, const Tensor<CPUBackend> &ref) {
+    EXPECT_EQ(img.shape(), ref.shape()) << "Different shapes";
+    auto img_view = view<const OutputType>(img), ref_view = view<const OutputType>(ref);
+    for (int i = 0; i < volume(img.shape()); i++) {
+      // TODO(skarpinski) Pretty-print position on error
+      ASSERT_EQ(img_view.data[i], ref_view.data[i]) << "at " << PrettyPosition(i, img.shape());
+    }
```

Comment:
```suggestion
    Check(view<const OutputType>(img), view<const OutputType>(ref));
```

from `dali/test/tensor_test_utils.h`

Reply:
Wow, thanks!

Reply:
Done